### PR TITLE
Update analyzer dep to 0.27.4-alpha.8.

### PR DESCRIPTION
**Pull request by pq**
Allows us to:
- use new EmbedderSdk (a step towards using SDK summaries)
- enjoy analyzer perf improvements (https://codereview.chromium.org/2011183004/)

### CODE REVIEW

**Comment by abarth** [comment]
LGTM

File: packages/flutter_tools/lib/src/dart/analysis.dart [hunk 1]
```diff
@@ -90,22 +90,20 @@ class AnalysisDriver {
...
     yamlLocator.refresh(packageMap);
 
-    EmbedderUriResolver embedderUriResolver =
-        new EmbedderUriResolver(yamlLocator.embedderYamls);
-    if (embedderUriResolver.length == 0) {
-      resolvers.add(new DartUriResolver(sdk));
+    EmbedderSdk sdk = new EmbedderSdk(yamlLocator.embedderYamls);
+    sdk.analysisOptions = context.analysisOptions;
+    // TODO(pq): support embedder summaries (https://github.com/dart-lang/sdk/issues/26467).
+    //sdk.useSummary = false;
```
Comment (devoncarew) [inline packages/flutter_tools/lib/src/dart/analysis.dart]: just to confirm, this defaults to false?

**Comment by devoncarew** [comment]
lgtm!

**Comment by pq** [comment]
Holding off on merging pending resolution of this API breakage issue: https://github.com/dart-lang/sdk/issues/26562.

File: packages/flutter_tools/lib/src/dart/analysis.dart (hunk 1 above)
Comment (pq) [inline packages/flutter_tools/lib/src/dart/analysis.dart]: Yep.

**Comment by pq** [comment]
Closing in favor of a rework w/ a new alpha with restored embedder API (https://github.com/dart-lang/sdk/issues/26562).

**Pull request closed without merging**